### DISCUSSION TITLE
What is the best class for a flyout?

### DISCUSSION THREAD

**Discussion by AkazaRenn** [dotMorten/WinUIEx#148] [Q&A] [answered]
I'm trying to implement a flyout that will appear on the screen by some shortcuts, similar to new virtual desktop indicator in 23H2.

![1682930790_virtual_desktops.gif](https://github.com/dotMorten/WinUIEx/assets/8340896/a789491a-8aba-4b57-bc61-d8ab54c24f91)

Currently my idea is using a window with the title bar and all UI buttons removed, so it would appear like a flyout. However, the window class doesn't seem to support adjusting its size automatically to fit the content.

Is there any suggestions on what would be the best way to implement a flyout in WinUI?

**Comment by dotMorten** [comment]
Why can't you adjust the size of the window to the size of your content?

**Reply by AkazaRenn** [reply]
> Why can't you adjust the size of the window to the size of your content?

Thank you for your reply! Just to confirm, what you mean is to do the resizing by code, instead of hoping it to fit the contents automatically, is that correct?

The content will be a text block with varying texts, may I ask if there's a way to get the width needed by that text block with a set of given string, font family and font size?

**Reply by dotMorten** [reply, marked as answer]
You can make your own control that overrides measure and arrange and looks at the desired size of a textblock. Use the desired size to make an informed decision about how large you want your window parent to be. This is a bit beyond what this discussion forum is about though.

**Reply by AkazaRenn** [reply]
I would definitely give it a try. Thank you!

**Reply by AkazaRenn** [reply]
> You can make your own control that overrides measure and arrange and looks at the desired size of a textblock. Use the desired size to make an informed decision about how large you want your window parent to be. This is a bit beyond what this discussion forum is about though.

Sorry for bothering again. I was playing with the window and realized that windows always have a border that is tinted by the OS's theme color, that would make it very unlike an actual flyout. Do you know if there's a way to remove the window border?

**Reply by AkazaRenn** [reply]
Sorry nevermind, just found the discussion at the WinUI repo: https://github.com/microsoft/microsoft-ui-xaml/issues/1247
